addon/-private/system/relationships/state/has-many.js

```diff
@@ -112,13 +111,13 @@ ManyRelationship.prototype.reload = function() {
   }
 
   if (this.link) {
-    return this.fetchLink();
+    this._loadingPromise = promiseManyArray(this.fetchLink(), 'Reload with link');
+    return this._loadingPromise;
   } else {
-    return this.store.scheduleFetchMany(this.manyArray.toArray()).then(function() {
-      //Goes away after the manyArray refactor
-      self.manyArray.set('isLoaded', true);
-      return self.manyArray;
-    });
+    this._loadingPromise = promiseManyArray(this.store.scheduleFetchMany(this.manyArray.toArray()).then(() => {
+      return this.manyArray;
+    }), 'Reload with ids');
+    return this._loadingPromise;
```

Comment:
@pangratz would you mind adding a test to ensure this else case is working correctly as well?